generators/Input/JTokenHelper.cs

```diff
@@ -45,16 +45,12 @@ public static dynamic ConvertJToken(JToken jToken)
 
         private static dynamic ConvertJObject(JObject jObject)
         {
-            var properties = jObject.ToObject<IDictionary<string, dynamic>>();
```

Comment:
This line was the main issue, as the conversion here would result in the dictionary not containing `JToken` values, which meant that the `ConvertJToken` functionality (which already had code to deal with long/int value conversions) was not called.

Interestingly, the new code is way easier!